portal-test/src/com/liferay/portal/kernel/test/util/ServiceContextTestUtil.java

```diff
@@ -51,23 +52,25 @@ public static ServiceContext getServiceContext(long groupId)
 	public static ServiceContext getServiceContext(long groupId, long userId)
 		throws PortalException {
 
-		if (groupId == TestPropsValues.getGroupId()) {
-			return getServiceContext(
-				TestPropsValues.getCompanyId(), groupId, userId);
-		}
+		return getServiceContext(groupId, userId, new long[0], new String[0]);
+	}
 
-		Group group = GroupLocalServiceUtil.getGroup(groupId);
+	public static ServiceContext getServiceContext(
+		long companyId, long groupId, long userId) {
 
 		return getServiceContext(
-			group.getCompanyId(), group.getGroupId(), userId);
+			companyId, groupId, userId, new long[0], new String[0]);
 	}
 
 	public static ServiceContext getServiceContext(
-		long companyId, long groupId, long userId) {
+		long companyId, long groupId, long userId, long[] assetCategoryIds,
+		String[] assetTagNames) {
 
 		ServiceContext serviceContext = new ServiceContext();
 
 		serviceContext.setAddGroupPermissions(true);
+		serviceContext.setAssetCategoryIds(assetCategoryIds);
+		serviceContext.setAssetTagNames(assetTagNames);
 		serviceContext.setAddGuestPermissions(true);
```

Comment:
maybe setAddGuestPermissions should go before setAssetCategoryIds?

Reply:
Indeed!!